lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -69,6 +69,16 @@ const std::function<void(SingleValueInstruction *, SILValue)>
           i->eraseFromParent();
         };
 
+Optional<SILBasicBlock::iterator> swift::getInsertAfterPoint(SILValue val) {
+  if (isa<SingleValueInstruction>(val)) {
+    return std::next(cast<SingleValueInstruction>(val)->getIterator());
+  }
+  if (isa<SILArgument>(val)) {
+    return cast<SILArgument>(val)->getParentBlock()->begin();
```

Comment:
This is just a quick drive by, but there is an API that already supports this: getDefiningInsertionPt.

Reply:
Hmmm... I see why this is a little different.